includes/class-search.php

```diff
@@ -0,0 +1,154 @@
+<?php
+/**
+ * ActivityPub Search Enhancement Class
+ *
+ * @package Activitypub
+ */
+
+namespace Activitypub;
+
+use Activitypub\Collection\Interactions;
+
+/**
+ * ActivityPub Search Enhancement Class
+ *
+ * This class enhances WordPress search functionality to detect URLs in search queries
+ * and attempt to import ActivityPub objects if found, otherwise falls back to classic search.
+ */
+class Search {
+
+	/**
+	 * Initialize the search enhancement.
+	 */
+	public static function init() {
+		\add_filter( 'pre_get_posts', array( self::class, 'enhance_public_search' ) );
+		\add_action( 'admin_init', array( self::class, 'enhance_admin_comment_search' ) );
```

Comment:
```suggestion
		\add_action( 'load-edit-comments.php', array( self::class, 'enhance_admin_comment_search' ) );
```
Changing it to the load hook would allow you to remove most checks from `enhance_admin_comment_search()`. Should this live in wp-admin somewhere?